libkbfs/disk_block_cache_test.go

```diff
@@ -758,3 +760,51 @@ func TestDiskBlockCacheMoveBlock(t *testing.T) {
 	require.Equal(t, 1, cache.syncCache.numBlocks)
 	require.Equal(t, TriggeredPrefetch, prefetchStatus)
 }
+
+func TestDiskBlockCacheHomeDirPriorities(t *testing.T) {
+	t.Parallel()
+	t.Log("Test that blocks from a home directory aren't evicted when there" +
+		" are other options.")
+	cache, config := initDiskBlockCacheTest(t)
+	defer shutdownDiskBlockCacheTest(cache)
+
+	ctx := context.Background()
+
+	t.Log("Set home directories on the cache")
+	homeTLF := tlf.FakeID(11, tlf.Private)
+	cache.AddHomeTLF(ctx, homeTLF)
+	homePublicTLF := tlf.FakeID(11, tlf.Public)
+	cache.AddHomeTLF(ctx, homePublicTLF)
+
+	t.Log("Seed the cache with blocks")
+	numTlfs := 12
+	numBlocksPerTlf := 5
+	numBlocks := numTlfs * numBlocksPerTlf * 2
+	seedDiskBlockCacheForTest(t, ctx, cache, config, numTlfs,
+		numBlocksPerTlf, tlf.Private)
+	seedDiskBlockCacheForTest(t, ctx, cache, config, numTlfs,
+		numBlocksPerTlf, tlf.Public)
+
+	countToEvict := numBlocks - numBlocksPerTlf*2 - 2
+	t.Log("Evict most of the blocks in the cache")
+	_, _, err := cache.syncCache.evictLocked(ctx, countToEvict)
+	require.NoError(t, err)
+	require.Equal(t, numBlocksPerTlf, cache.syncCache.tlfCounts[homeTLF])
+	require.Equal(t, numBlocksPerTlf, cache.syncCache.tlfCounts[homePublicTLF])
+
+	// evict the last 2 non-home-dir blocks and n-2 public homedir blocks
+	countToEvict = numBlocksPerTlf
+	t.Log("Evict most of the remaining blocks in the cache")
+	_, _, err = cache.syncCache.evictLocked(ctx, countToEvict)
+	require.NoError(t, err)
+	require.Equal(t, numBlocksPerTlf, cache.syncCache.tlfCounts[homeTLF])
+	require.Equal(t, 2, cache.syncCache.tlfCounts[homePublicTLF])
+
+	// evict the last 2 public-home-dir blocks and n-2 private homedir blocks
+	countToEvict = numBlocksPerTlf
+	t.Log("Evict most of the remaining blocks in the cache")
+	_, _, err = cache.syncCache.evictLocked(ctx, countToEvict)
+	require.NoError(t, err)
+	require.Equal(t, 2, cache.syncCache.tlfCounts[homeTLF])
```

Comment:
IMO you're on the right track for this test, but you should explicitly have these be true:

1. Make sure ever TLF is seeded, including the 2 home TLFs.
2. Distribute the blocks exponentially over the non-home TLFs. Use LOTS of blocks to get better statistical behavior.
3. Evict half the non-home TLF blocks using small eviction sizes (each eviction is 10% of the starting size of the _smallest_ TLF in your distribution). (tlf sizes 100, 200, 400, 800, 1600, eviction size 10). Verify that the non-home TLFs have been reduced in size by about half at the end of this step. (give or take 20%, so 30-70%)
4. Evict the rest of the non-home TLF blocks in ~2 evictions.
5. Evict enough blocks to get rid of the public home TLF.
6. Evict enough blocks to get rid of the private home TLF.

Reply:
If step 2 leaves you a bit flaky, run the whole thing 3 times and take the average.